src/mono/mono/metadata/exception.c

```diff
@@ -870,6 +870,49 @@ mono_get_exception_type_initialization_handle (const gchar *type_name, MonoExcep
 	HANDLE_FUNCTION_RETURN_REF (MonoException, MONO_HANDLE_CAST (MonoException, exc));
 }
 
+MonoExceptionHandle
+mono_get_exception_thread_start_handle (MonoException* inner_raw, MonoError *error)
```

Comment:
Looks ok.  It might be better to invoke a managed method that creates the exception and throws.
```
  void ThrowThreadStartException (Exception inner) => throw new ThreadStartException (inner);
```
and propagate the `MonoError` out to the caller in the native code.